ADR/ADR015-use-paper-trail-for-draft-live-or-some-other-solution.md

```diff
@@ -0,0 +1,57 @@
+that deals with [associations](https://github.com/paper-trail-gem/paper_trail#4b-associations).
+This gem should be able to give us the  functionality we needed such as trying to identify which versions of pages make 
+up a specific version of the form.
+
+### Positives
+- PT-AT can identify which forms/pages where created/updated and it was possible to restore that use to reuse
+- PT allows custom events so a new event for "make live" could be used to identify specific live versions.
+- Keep audit of events and records together in one table.
+
+### Negatives
+- PT-AT doesn't work well when deleting associations such as pages and it was found that .length/size/count after delete
+  didn't return the expected number of pages df a form as it seemed to query the pages table rather than version
+- PT-AT contributor recommends not using PT-AT because of it dependency on PT and its "blackbox" approach which doesn't seem
+  right to just assume everything works.
+
+## Decision
+
+The team have decided not to implement draft/live versions with papertrail for the following reasons
+
+- We cannot safely delete pages and be able to retrieve the specific live form which may have used that delete page.
+- There are a lot of [limitations](https://github.com/westonganger/paper_trail-association_tracking#limitations) and known issues  (https://github.com/westonganger/paper_trail-association_tracking#known-issues). 
+  One such issue that we came across "Not compatible with transactional tests" which made it extremely hard if not possible
+  to test any data we were expecting.
+
+The PT-AT gem states the following as alternative solution
+
+> Model Versioning and Restoration require concious thought, design, and understanding. You should understand your versioning and restoration process completely. Because PT-AT it is mostly a blackbox solution which encourages you to set it up and then assume its "Just Working". This can make for major data problems later.
+> 
+> Instead I recommend a newer gem that I have created for handling snapshots of records and associations called [active_snapshot](https://github.com/westonganger/active_snapshot). This gem does not utilize paper_trail at all. The focus of this gem is to have a simple and fully understandable design is easy to customize and know inside and out for your projects needs.
+
+We did briefly investigate using acive_snapshot but there were a couple of issues and concerns we had about using it
+
+- This suggestion was made over 2 years ago by the author of PT-AT which is why they are recommencing it. Looking at downloads
+  from RubyGem.org (at this time, theres been a total of 18,858 downloads of which 1893 where for the current version 0.3.0).
+  This seems rather low given the time period and also not main blog posts/mentions of it online.
+- We would have to use this in addition to papertrail as it isn't automatically integrated into active record create,update,read,delete
```

Comment:
As with the PT-AT reasoning, think these could be covered in the Context section as negatives.